Formula/gap.rb

```diff
@@ -0,0 +1,70 @@
+class Gap < Formula
+  desc "System for computational discrete algebra"
+  homepage "https://www.gap-system.org/"
+  url "https://www.gap-system.org/pub/gap/gap-4.9/tar.bz2/gap-4.9.1.tar.bz2"
+  version "4.9.1"
+  sha256 "d93abb263e6b9ce9f56bfb820ac263b9a0f349f6bd1584244c7b242931d0dd3a"
+
+  depends_on "gmp"
+  depends_on "readline"
```

Comment:
They asked about why GNU readline is needed in their previous review, so perhaps add a comment here like this:
```
# GAP cannot be built against the native OSX version of readline, it requires either GNU readline, or
# no readline at all; but the latter leads to an inferior user experience. So we depend on GNU readline here.
```